prusti-viper/src/encoder/middle/core_proof/types/interface.rs

```diff
@@ -172,7 +172,9 @@ impl<'p, 'v: 'p, 'tcx: 'v> Private for Lowerer<'p, 'v, 'tcx> {
                 } else {
                     let parameters = vars! {
                         address: Address,
-                        target_current: {target_type.clone()}
+                        target_current: {target_type.clone()},
+                        // NOTE: final destructor is also necessary for shared references
```

Comment:
Why is this the case?

Reply:
The assign method did not distinguish between shared and mutable references. Should be ok now.

This is for shared references:

```
  // bb0[4] _2 = &'_#3r _1
  tmp$0 := assign$ref$Shared$I32$ref_$I32(_2$place(), _2$address, _1$place(), _1$address, _1$snapshot$1, lft_3$snapshot$0, write / 3)
  inhale _2$snapshot$1 == tmp$0
  inhale _1$snapshot$2 == destructor$Snap$ref$Shared$I32$$target_current(tmp$0)
```